src/main/java/core/basesyntax/service/io/impl/CsvDataReader.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service.io.impl;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import core.basesyntax.model.entities.Product;
+import core.basesyntax.model.entities.exception.InvalidFileExtensionException;
+import core.basesyntax.service.io.DataReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class CsvDataReader<T extends Product> implements DataReader<T> {
+    private final String inputFilePath;
```

Comment:
we should better have this as the method param, so that we could pass different paths to be read